src/index.ts

```diff
@@ -34,21 +34,26 @@ function unmount() {
 }
 
 function startObserving() {
-  stopObserving();
-  if (!location.hash) return;
+  cancelAnimationFrame(throttleStartObservingId);
 
-  addEventListener("touchend", stopObserving);
-  addEventListener("wheel", stopObserving);
-  document.addEventListener("selectstart", stopObserving);
-  documentObserver?.observe(document, OBSERVER_CONFIG);
-  adjustScrollPosition();
+  throttleStartObservingId = requestAnimationFrame(() => {
```

Comment:
We should avoid anonymous functions in a "Do not use more resources then needed" PR. 😉 

* [ ] Please name the function